File: app/src/main/java/com/railian/maksym/provectustesttask/repository/DatabaseHelper.java
package com.railian.maksym.provectustesttask.repository;

import android.util.Log;
import com.raizlabs.android.dbflow.config.FlowManager;
import com.raizlabs.android.dbflow.sql.language.Select;
import com.raizlabs.android.dbflow.structure.BaseModel;
import com.raizlabs.android.dbflow.structure.database.transaction.ProcessModelTransaction;
import com.raizlabs.android.dbflow.structure.database.transaction.Transaction;

import java.util.List;

public class DatabaseHelper {
    public void insert(List<DatabaseModel> items) {
        FlowManager.getDatabase(Database.class).beginTransactionAsync(new ProcessModelTransaction.Builder<>(
                (ProcessModelTransaction.ProcessModel<DatabaseModel>) BaseModel::save).addAll(items).build()).error((transaction, error) -> Log.e("FAILED SELECT: " + transaction, error.toString())).success(new Transaction.Success() {
            @Override
            public void onSuccess(Transaction transaction) {
                Log.e("SUCESS SELECT", transaction.toString());
            }
        }).build().execute();
    }


    public List<DatabaseModel> select() {
        return new Select().from(DatabaseModel.class).queryList();
    }

    public DatabaseModel selectByID(int id) {
        return (new Select().from(DatabaseModel.class).where(DatabaseModel_Table.Id.is(id + 1)).querySingle());
    }

    public void deleteAll() {
        FlowManager.getDatabase(Database.class).beginTransactionAsync(new ProcessModelTransaction.Builder<>(
                (ProcessModelTransaction.ProcessModel<DatabaseModel>) BaseModel::delete).addAll(select()).build()).error((transaction, error) -> Log.e("FAILED FELETE: " + transaction, error.toString())).success(new Transaction.Success() {
            @Override
            public void onSuccess(Transaction transaction) {
                Log.e("SUCESS DELETE", transaction.toString());
            }
        }).build().execute();
    }
}

File: app/src/main/java/com/railian/maksym/provectustesttask/repository/DatabaseHelper.kt
package com.railian.maksym.provectustesttask.repository

import android.arch.persistence.room.*
import android.arch.persistence.room.OnConflictStrategy.REPLACE


@Dao interface DatabaseHelper {
    @Query("select * from DatabaseModel")
    fun getAllTasks(): List<DatabaseModel>

    @Query("select * from DatabaseModel where id = :id")
    fun findTaskById(id: Int): DatabaseModel

    @Insert(onConflict = REPLACE)
    fun insertTask(task: DatabaseModel)

    @Update(onConflict = REPLACE)
    fun updateTask(task: DatabaseModel)

    @Delete
    fun deleteTask(task: DatabaseModel)

    @Query("DELETE FROM DatabaseModel")
    fun dropTable()

    @Insert(onConflict = REPLACE)
    fun addAll(objects: List<DatabaseModel>)
}